lib/ansible/modules/cloud/azure/azure_rm_image.py

```
@@ -0,0 +1,372 @@
+            disk = disk_instance.id
+        return (blob_uri, disk, snapshot)
+
+    def create_os_disk(self):
+        blob_uri, disk, snapshot = self.resolve_storage_source(self.source)
+        snapshot_resource = SubResource(snapshot) if snapshot else None
+        managed_disk = SubResource(disk) if disk else None
+        return ImageOSDisk(os_type=self.os_type,
+                           os_state=OperatingSystemStateTypes.generalized,
+                           snapshot=snapshot_resource,
+                           managed_disk=managed_disk,
+                           blob_uri=blob_uri)
+
+    def create_data_disk(self, lun, source):
+        blob_uri, disk, snapshot = self.resolve_storage_source(source)
+        if blob_uri or disk or snapshot:
+            snapshot_resource = SubResource(snapshot) if snapshot else None
+            managed_disk = SubResource(disk) if disk else None
+            return ImageDataDisk(lun,
+                                 blob_uri=blob_uri,
+                                 snapshot=snapshot_resource,
+                                 managed_disk=managed_disk)
+
+    def create_data_disks(self):
+        return list(filter(None, [self.create_data_disk(lun, source) for lun, source in enumerate(self.data_disk_sources)]))
+
+    def get_source_vm(self):
+        vm_resource_id = format_resource_id(self.source,
+                                            self.subscription_id,
+                                            'Microsoft.Compute',
+                                            'virtualMachines',
+                                            self.resource_group)
+        resource = parse_resource_id(vm_resource_id)
+        return self.get_vm(resource['resource_group'], resource['name']) if resource['type'] == 'virtualMachines' else None
+
+    def get_snapshot(self, snapshot_name):
+        try:
+            return self.compute_client.snapshots.get(self.resource_group, snapshot_name)
+        except Exception:
```

Blanket exception-swallowing like this is dangerous and could mask a number of other issues... Should probably just catch `CloudError` and check if `error.error` is `ResourceNotFound`- if it's not, `raise` and be done.

---

Agree with u, and I will change that

---

It should check the `CloudError.error.error` is `ResourceNotFound`. I check the `status_code == 404` instead